resources/types/FBAModel.yaml

```diff
@@ -0,0 +1,46 @@
+global-object-type: FBAModel
+ui-type-name: FBA Model
+storage-type: WS
+storage-object-type: KBaseFBA.FBAModel
+versions:
+- indexing-rules:
+  - path: id
+    keyword-type: string
+  - path: name
+    full-text: true
+  - path: source
+    full-text: true
+  - path: type
+    keyword-type: string
+  - path: modelcompartments/{size}
+    keyword-type: integer
+    ui-name: Model Compartments
+  - path: modelcompounds/{size}
+    keyword-type: integer
+    ui-name: Model Compounds
+  - path: modelreactions/{size}
+    keyword-type: integer
+    ui-name: Model Reactions
+  - path: genome_ref
+    transform: guid
+    target-object-type: Genome
+    target-object-type-version: 1
+    optional-default-value: []
+    keyword-type: string
+    key-name: genome_guid
+    ui-hidden: true
+  - source-key: genome_guid
+    transform: lookup.key.scientific_name
+    full-text: true
+    key-name: scientific_name
+    ui-name: Genome Name
+  - source-key: genome_guid
+    transform: lookup.key.taxonomy
+    full-text: true
+    key-name: taxonomy
+    ui-name: Genome Taxonomy
+  - source-key: genome_guid
```

Comment:
Is the genome id mutable?

Reply:
It's as mutable as any other field in an object I think

Reply:
So it's immutable?

Reply:
Let me ask a different question - what is the genome id and where does it come from?

Reply:
Genome ID is a string that generally comes from the user when the genome is grated by upload or an app

Reply:
Is it the same as the workspace object name?

Reply:
Yes it is

Reply:
Ok, this is something I've never understood. The ID is the object name, which can be changed, but is embedded in the object, which can't be changed. That seems incredibly confusing. I removed that from the index for Assembly (IIRC) due to this issue. Do you know why the design this way? It seems like it's bound to cause user confusion (and I know testers have pointed it out as a bug before).

Reply:
Yeah, I see your point. Is there anyway to index the object name from a guid?

Reply:
There is, but there's no way for that object name to be updated, so it's just as bad. The search records generated from the specs are assumed to be immutable.

Reply:
To be clear, the object name of the given object being indexed is also indexed and is updated with changes. If it's specified in a search spec that's not the case.

Reply:
Given those constraints, I think it's still better to leave the id in. Sure it might confuse some users if they rename a genome and it still pops up in the search but I suspect this is a much smaller issue than for users not to be able to find metabolic models generated from a given genome.

Reply:
Hmm, I predict bug reports in our future, so I'd like to get a call on this from @sychan or @scanon

Reply:
scanon [11:17 AM]
I think we should keep it for now.

Better to have the search option even if it sometimes would be out of sync than to totally drop it.

Reply:
Where is this message? I'm not able to find it in slack.

Reply:
It was a DM with Shane. We can just discuss at standup

Reply:
Confirmed with Shane